docs/site/pages/dsl/views.mdx

```diff
@@ -0,0 +1,128 @@
+---
+title: Writing DSL Views
+---
+
+# Overview
+Writing assets or views is as simple as creating a React element using your base DSL components:
+
+```tsx
+import { Input, Text, Collection } from '@player-ui/reference-assets-components';
+
+const view = (
+  <Collection>
+    <Text>Some value</Text>
+    <Input>
+      <Input.Label>Some label</Input.Label>
+    </Input>
+  </Collection>
+);
+```
+
+When compiled, this would produce the following JSON.
+
+```json
+{
+  "id": "root",
+  "type": "collection",
+  "values": [
+    {
+      "asset": {
+        "id": "root-values-1",
+        "type": "text",
+        "value": "Some value"
+      }
+    },
+    {
+      "asset": {
+        "id": "root-values-2",
+        "type": "input",
+        "label": {
+          "asset": {
+            "id": "root-values-2-label",
+            "type": "text",
+            "value": "Some label"
+          }
+        }
+      }
+    }
+  ]
+}
+```
+
+Not only is the source DSL content a fraction of the output object's size (making it easier to read and maintain) as the base components use the same TypeScript types as the assets themselves, you will recieve in editor suggestions and type checks as you author your content. 
+
+# View Concepts in DSL
+
+## Templates
+
+Templates are included via the `@player-tools/dsl` package. This can be used in any asset slot:
+
+```tsx
+import { dataTypes } from '@player-ui/common-types-plugin';
+import { makeBindingsForObject, Template } from '@player-tools/dsl';
+
+const schema = {
+  foo: [{
+    bar: dataTypes.StringType,
+  }],
+};
+
+const bindings = makeBindingsForObject(schema);
+
+<Collection>
+  <Collection.Values>
+    <Template data={bindings.foo}>
+      <Text>The current item is: {bindings.foo._index_.bar}</Text>
+    </Template>
+  </Collection.Values>
+</Collection>
+```
+
+Templates can be nested within one another, and the auto-id generation will handle adding the `_index_` information to any generated `id`.
+
+## Switches
+
+The `@player-tools/dsl` module also includes support for _static_ and _dynamic_ switches.
+
+Use the `isDynamic` flag to denote this should be a `dynamicSwitch` instead of a `staticSwitch`:
+
+```tsx
+import { Switch } from '@player-ui/dsl';
+
+<Collection>
+  <Collection.Label>
+    <Switch isDynamic>
+      <Switch.Case exp={e`foo() = bar()`}>
+        <Text>Text 1</Text>
+      </Switch.Case>
+      <Switch.Case>
+        <Text>Text 1</Text>
+      </Switch.Case>
+    </Switch>
+  </Collection.Label>
+</Collection>
+```
+
+# DSL Benefits in Views
```

Comment:
Is it worth doing a jump link from the DSL intro to this section? Conceptually I feel like these are both benefits of the DSL, but also the mindset to approach writing your own components.

Reply:
Actually I think you may be right. I might write a higher level benefits section in the DSL info page but link out to specific sections in the views/schema pages for more detailed explanations of features?